front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx

```diff
@@ -0,0 +1,238 @@
+        crop.y ? (crop.y / 100) * image.naturalHeight : 0,
+        canvas.width,
+        canvas.height,
+        0,
+        0,
+        canvas.width,
+        canvas.height
+      );
+    }
+
+    return new Promise((resolve, reject) => {
+      canvas.toBlob((blob) => {
+        if (!blob) {
+          reject(new Error("Canvas is empty"));
+          return;
+        }
+        blob = new File([blob], "avatar.jpeg", { type: "image/jpeg" });
+        const fileUrl = window.URL.createObjectURL(blob);
+        resolve(fileUrl);
+      }, "image/jpeg");
+    });
+  };
+
+  const onFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const file = e?.target?.files?.[0];
+    if (!file) return;
+    setCurrentTab("upload");
+    const reader = new FileReader();
+    reader.addEventListener("load", () => setSrc(reader.result as string));
+    reader.readAsDataURL(file);
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={onClose}
+      title="Pick the design of your assistant:"
+      isFullScreen={false}
+      hasChanged={currentTab === "upload" && !!src}
+      onSave={isUploadingAvatar ? undefined : onUpload}
+    >
+      <input
+        type="file"
+        style={{ display: "none" }}
+        onChange={onFileChange}
+        ref={fileInputRef}
+      />
+      <div className="pt-8">
+        <Tab
+          tabs={[
+            {
+              label: "Pick from the library",
+              current: currentTab === "pick",
+              icon: ImageIcon,
+            },
+            {
+              label: "Upload your own",
+              current: currentTab === "upload",
+              icon: ArrowUpOnSquareIcon,
+            },
+          ]}
+          onTabClick={(tab) => {
+            const isUploadTab = tab.startsWith("Upload");
+            if (isUploadTab) {
+              // triggers file selection. once file is selected, tab will change to `upload`
+              fileInputRef?.current?.click();
+            } else {
+              setCurrentTab("pick");
+            }
+          }}
+        />
+      </div>
+      {currentTab === "pick" && (
+        <div className="grid max-w-3xl grid-cols-8 gap-4 pt-8">
+          {avatarUrls.map(({ available, url }) => (
+            <div
+              key={url}
+              className={classNames(
+                available ? "cursor-pointer" : "opacity-30"
+              )}
+              onClick={() => {
+                if (available) {
+                  onPick(url);
+                  onClose();
+                }
+              }}
+            >
+              <Avatar
+                size="lg"
+                visual={<img src={url} />}
+                clickable={available}
+              />
+            </div>
+          ))}
+        </div>
+      )}
+      {currentTab === "upload" && src && (
+        <div className="pt-8">
+          <ReactCrop crop={crop} aspect={1} onChange={(_, pC) => setCrop(pC)}>
```

Comment:
Should be centered? :grimacing:

Reply:
😬

Reply:
Fixed

Reply:
<img width="1785" alt="Screenshot 2023-09-26 at 15 26 46" src="https://github.com/dust-tt/dust/assets/14199823/8377494f-d073-4a2f-a38a-38d32fb6e691">